web/client/observables/autocomplete.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+ /**
+ * Observables used for autocomplete in the feature editor
+ * @name observables.autocomplete
+ * @type {Object}
+ *
+*/
+const Rx = require('rxjs');
+const axios = require('../libs/ajax');
+const {getWpsPayload} = require('../utils/ogc/WPS/autocomplete');
+
+/**
+ * creates a stream for fetching data via WPS
+ * @param  {external:Observable} props
+ * @memberof observables.autocomplete
+ * @return {external:Observable} the stream used for fetching data for the autocomplete editor
+*/
+const createPagedUniqueAutompleteStream = (props$) => props$.debounce(props => Rx.Observable.timer(props.delayDebounce || 0))
```

Comment:
You can improve with: 
```
    .distinctUntilChanged( ({value, currentPage}, newProps = {}) => !(newProps.value !== value || newProps.currentPage !== currentPage))
    // .debounce(props => Rx.Observable.timer(props.delayDebounce || 0))
    .throttle(props => Rx.Observable.timer(props.delayDebounce || 0))
    .merge(props$.debounce(props => Rx.Observable.timer(props.delayDebounce || 0))).distinctUntilChanged()
```
This will hide quickly "No Result" message and perform a faster search (start with first insert, then throttle and emit last). Also change default time to 500